lsr-role2collection.py

```diff
@@ -265,19 +265,12 @@ def file_replace(path, find, replace, file_patterns):
                     f.write(s)
 
 
-def replace_rolepath_with_collection(namespace, collection, role):
-    """
-    Replace "{{ role_path }}/roles/rolename" with namespace.collection.rolename
-    in role_dir.
-    """
-    role_dir = output / 'roles' / role
-    find = "{{ role_path }}/roles/(.*)"
-    replace = namespace + "." + collection + ".\\1"
-    file_patterns = ['*.yml', '*.md']
-    file_replace(role_dir, find, replace, file_patterns)
-
-# Replace "{{ role_path }}/roles/ROLE" with NAMESPACE.COLLECTION.ROLE
-replace_rolepath_with_collection(namespace, collection, role)
+# Replace "{{ role_path }}/roles/rolename" with rolename in role_dir.
+role_dir = output / 'roles' / role
+find = "{{ role_path }}/roles/(.*)"
+replace = ".\\1"
+file_patterns = ['*.yml', '*.md']
+file_replace(role_dir, find, replace, file_patterns)
```

Comment:
What about cases where the test has something like this:
```yaml
include_tasks: "{{ role_path }}/roles/rolename/vars/main.yml"
```
?

Reply:
This is a sub-role case as `rsyslog` in `logging`.
`include_tasks: "{{ role_path }}/roles/rolename/tasks/main.yml"`
So, rolename is promoted to the same level as the role that includes the task.
I think this'd work. 
`include_tasks: "NAMESPACE.COLLECTION.rolename/tasks/main.yml"`
But how about without `NAMESPACE.COLLECTION.`? If it is guaranteed, it is simpler and better.

Reply:
in the logging case, would it work if you replaced
`include_tasks: "{{ role_path }}/roles/rsyslog/tasks/main.yml"`
with
`include_tasks: "{{ role_path }}/../rsyslog/tasks/main.yml"`

or even
```yaml
include_role:
  name: fedora.system_roles.rsyslog
```
?

Reply:
Logging role has only `include_role` which goes across the role boundary:
```
  include_role:
    name: "{{ role_path }}/roles/rsyslog"
```
It does not have `include_tasks` nor `include_vars` doing that. Maybe, I should add such test cases locally...

For `include_role`, I verified both of them work.
```
include_role:
  name: fedora.system_roles.rsyslog

include_role:
  name: rsyslog
```

Reply:
> What about cases where the test has something like this:
> 
> ```yaml
> include_tasks: "{{ role_path }}/roles/rolename/vars/main.yml"
> ```
> 
> ?

I suppose you meant `include_vars`, is there an example of this? The only one I recall was in kernel_settings and you fixed that. My suggested approach would be "don't do it then" (it caused problems even outside the context of collections and users should not do anything similar, so a test doing this is unrealisitic).

Reply:
> It does not have include_tasks nor include_vars doing that. Maybe, I should add such test cases locally...

@pcahyna, Ok. As I wrote above, there's no case to cross the role boundary for `include_tasks` and `_vars`. We are going to drop supporting them.

Just for record, collections recommend to use the FQCN in include_ and import_tasks. 
```
https://docs.ansible.com/ansible/2.10/collections/ansible/builtin/import_tasks_module.html
https://docs.ansible.com/ansible/2.10/collections/ansible/builtin/include_tasks_module.html
This module (ansible.builtin.import_tasks,include_tasks) is part of ansible-base and 
included in all Ansible installations. In most cases, you can use the short module name 
import_tasks even without specifying the collection: keyword. Despite that, we recommend 
you use the FQCN for easy linking to the module documentation and to avoid conflicting 
with other collections that may have the same module name.
```